test-clients/src/main/java/com/hedera/services/bdd/suites/schedule/ScheduleExecutionSpecs.java

```diff
@@ -143,14 +158,515 @@ protected Logger getResultsLogger() {
+
+							Assert.assertEquals("Wrong record account ID!",
+									createTx.getResponseRecord().getTransactionID().getAccountID(),
+									triggeredTx.getResponseRecord().getTransactionID().getAccountID());
+
+							Assert.assertTrue("Transaction not scheduled!",
+									triggeredTx.getResponseRecord().getTransactionID().getScheduled());
+
+							Assert.assertEquals("Wrong schedule ID!",
+									createTx.getResponseRecord().getReceipt().getScheduleID(),
+									triggeredTx.getResponseRecord().getScheduleRef());
+						}),
+						getTokenInfo(A_TOKEN)
+								.hasTotalSupply(0)
+				);
+	}
+
+	private HapiApiSpec scheduledUniqueMintFailsWithInvalidMetadata() {
+		String failingTxn = "failingTxn";
+		return defaultHapiSpec("ScheduledUniqueMintFailsWithInvalidBatchSize")
+				.given(
+						overriding("scheduling.whitelist", "TokenMint"),
+						overriding("tokens.nfts.areEnabled", "true"),
+						cryptoCreate("treasury"),
+						cryptoCreate("schedulePayer"),
+						newKeyNamed("supplyKey"),
+						tokenCreate(A_TOKEN)
+								.supplyKey("supplyKey")
+								.treasury("treasury")
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0),
+						scheduleCreate("validSchedule",
+								mintToken(
+										A_TOKEN, List.of(metadataOfLength(101))
+								)
+						)
+								.designatingPayer("schedulePayer")
+								.via(failingTxn)
+				)
+				.when(
+						scheduleSign("validSchedule")
+								.alsoSigningWith("supplyKey", "schedulePayer", "treasury")
+								.hasKnownStatus(SUCCESS)
+				)
+				.then(
+						getTxnRecord(failingTxn).scheduled()
+								.hasPriority(recordWith().status(METADATA_TOO_LONG)),
+						getTokenInfo(A_TOKEN)
+								.hasTotalSupply(0)
+				);
+	}
+
+	private HapiApiSpec scheduledUniqueBurnFailsWithInvalidNFTID() {
+		String failingTxn = "failingTxn";
+		return defaultHapiSpec("ScheduledUniqueBurnFailsWithInvalidNFTID")
+				.given(
+						overriding("scheduling.whitelist", "TokenBurn"),
+						overriding("tokens.nfts.areEnabled", "true"),
+						cryptoCreate("treasury"),
+						cryptoCreate("schedulePayer"),
+						newKeyNamed("supplyKey"),
+						tokenCreate(A_TOKEN)
+								.supplyKey("supplyKey")
+								.treasury("treasury")
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0),
+						scheduleCreate("validSchedule",
+								burnToken(
+										A_TOKEN, List.of(123L)
+								)
+						)
+								.designatingPayer("schedulePayer")
+								.via(failingTxn)
+				)
+				.when(
+						scheduleSign("validSchedule")
+								.alsoSigningWith("supplyKey", "schedulePayer", "treasury")
+								.hasKnownStatus(SUCCESS)
+				)
+				.then(
+						getTxnRecord(failingTxn).scheduled()
+								.hasPriority(recordWith().status(INVALID_NFT_ID))
+				);
+	}
+
+	private HapiApiSpec scheduledBurnFailsWithInvalidAmount() {
```

Comment:
I think that a better test will be providing a normal positive number and expecting that it will fail for non-fungible uniques. We can leave this test but add the one proposed as-well